libcaf_net/test/endpoint_manager.cpp

```diff
@@ -107,8 +100,11 @@ class dummy_transport {
   template <class Manager>
   bool handle_write_event(Manager& mgr) {
     for (auto x = mgr.next_message(); x != nullptr; x = mgr.next_message()) {
-      auto& payload = x->payload;
-      buf_.insert(buf_.end(), payload.begin(), payload.end());
+      std::vector<byte> payload_buf;
+      binary_serializer sink{mgr.system(), payload_buf};
```

Comment:
Why the extra buffer instead of just serializing into `buf_`?

Reply:
I wasn't sure wether the serializer would simply append the data to the contained buffer.